tests/flytekit/unit/core/test_type_engine.py

```diff
@@ -967,6 +967,7 @@ class TestFileStruct(DataClassJsonMixin):
         b: typing.Optional[FlyteFile]
         b_prime: typing.Optional[FlyteFile]
         c: typing.Union[FlyteFile, None]
+        c_prime: typing.Union[None, FlyteFile]
```

Comment:
Although we didn’t expect `typing.Union[None, StructuredDataset, int, FlyteFile]` to work in our tests due to multiple FlyteTypes ambiguity, we anticipated something like `typing.Union[None, int, FlyteFile]` might function correctly. However, after tracing the code, I discovered that there are some bugs while msgpack handling unions with more than two types. Maybe it should be fixed in another PR? cc @Future-Outlier